### ISSUE TITLE
PanicException: index out of bounds: the len is 1 but the index is 1

### ISSUE THREAD

**Issue by useffc** [pierogis/pierogis#17]
input:
`RUST_BACKTRACE=full pyrogis threshold ./18deg.png -u 10 -l 10`

output:
```
cooking './18deg.png' to 'cooked.png'
thread '<unnamed>' panicked at 'index out of bounds: the len is 1 but the index is 1', src/lib.rs:77:20
stack backtrace:
   0:        0x102975908 - <std::sys_common::backtrace::_print::DisplayBacktrace as core::fmt::Display>::fmt::hb7980d26174b99cc
   1:        0x10298a8fc - core::fmt::write::hce37a9344a9aa8e7
   2:        0x102973b8c - std::io::Write::write_fmt::h686eb8440ccc31c3
   3:        0x102977410 - std::panicking::default_hook::{{closure}}::hd8ecce238a238901
   4:        0x1029770e8 - std::panicking::default_hook::h4c3bbdaeee039651
   5:        0x1029779c4 - std::panicking::rust_panic_with_hook::h140cbf507d3407f4
   6:        0x1029775a0 - std::panicking::begin_panic_handler::{{closure}}::hb6a3220cf34e266a
   7:        0x102975dc0 - std::sys_common::backtrace::__rust_end_short_backtrace::hd5353dc16fba6ea5
   8:        0x102977508 - _rust_begin_unwind
   9:        0x102990a90 - core::panicking::panic_fmt::h0333f3235caab578
  10:        0x102990a68 - core::panicking::panic_bounds_check::h6527e71421690366
  11:        0x102946ba4 - core::ops::function::impls::<impl core::ops::function::FnMut<A> for &F>::call_mut::hf658b551ca70d4b6
  12:        0x102946998 - rayon::iter::plumbing::bridge_producer_consumer::helper::h46dc11a49876a36b
  13:        0x10294f228 - rayon_core::job::StackJob<L,F,R>::run_inline::h6d87b8dede6b685a
  14:        0x102948628 - rayon_core::registry::in_worker::h4d824598a3d5b795
  15:        0x102946a24 - rayon::iter::plumbing::bridge_producer_consumer::helper::h46dc11a49876a36b
  16:        0x10294f228 - rayon_core::job::StackJob<L,F,R>::run_inline::h6d87b8dede6b685a
  17:        0x102948628 - rayon_core::registry::in_worker::h4d824598a3d5b795
  18:        0x102946a24 - rayon::iter::plumbing::bridge_producer_consumer::helper::h46dc11a49876a36b
  19:        0x10294f228 - rayon_core::job::StackJob<L,F,R>::run_inline::h6d87b8dede6b685a
  20:        0x102948628 - rayon_core::registry::in_worker::h4d824598a3d5b795
  21:        0x102946a24 - rayon::iter::plumbing::bridge_producer_consumer::helper::h46dc11a49876a36b
  22:        0x10294f228 - rayon_core::job::StackJob<L,F,R>::run_inline::h6d87b8dede6b685a
  23:        0x102948628 - rayon_core::registry::in_worker::h4d824598a3d5b795
  24:        0x102946a24 - rayon::iter::plumbing::bridge_producer_consumer::helper::h46dc11a49876a36b
  25:        0x10294f228 - rayon_core::job::StackJob<L,F,R>::run_inline::h6d87b8dede6b685a
  26:        0x102948628 - rayon_core::registry::in_worker::h4d824598a3d5b795
  27:        0x102946a24 - rayon::iter::plumbing::bridge_producer_consumer::helper::h46dc11a49876a36b
  28:        0x10294f540 - <rayon_core::job::StackJob<L,F,R> as rayon_core::job::Job>::execute::h225507a10ea8fa97
  29:        0x10298f5f4 - rayon_core::registry::WorkerThread::wait_until_cold::h42bc99c4c3b80f32
  30:        0x102948648 - rayon_core::registry::in_worker::h4d824598a3d5b795
  31:        0x102946a24 - rayon::iter::plumbing::bridge_producer_consumer::helper::h46dc11a49876a36b
  32:        0x10294f540 - <rayon_core::job::StackJob<L,F,R> as rayon_core::job::Job>::execute::h225507a10ea8fa97
  33:        0x10298f5f4 - rayon_core::registry::WorkerThread::wait_until_cold::h42bc99c4c3b80f32
  34:        0x102948648 - rayon_core::registry::in_worker::h4d824598a3d5b795
  35:        0x102946a24 - rayon::iter::plumbing::bridge_producer_consumer::helper::h46dc11a49876a36b
  36:        0x10294f228 - rayon_core::job::StackJob<L,F,R>::run_inline::h6d87b8dede6b685a
  37:        0x102948628 - rayon_core::registry::in_worker::h4d824598a3d5b795
  38:        0x102946a24 - rayon::iter::plumbing::bridge_producer_consumer::helper::h46dc11a49876a36b
  39:        0x10294f228 - rayon_core::job::StackJob<L,F,R>::run_inline::h6d87b8dede6b685a
  40:        0x102948628 - rayon_core::registry::in_worker::h4d824598a3d5b795
  41:        0x102946a24 - rayon::iter::plumbing::bridge_producer_consumer::helper::h46dc11a49876a36b
  42:        0x10294f540 - <rayon_core::job::StackJob<L,F,R> as rayon_core::job::Job>::execute::h225507a10ea8fa97
  43:        0x10298f5f4 - rayon_core::registry::WorkerThread::wait_until_cold::h42bc99c4c3b80f32
  44:        0x102948648 - rayon_core::registry::in_worker::h4d824598a3d5b795
  45:        0x102946a24 - rayon::iter::plumbing::bridge_producer_consumer::helper::h46dc11a49876a36b
  46:        0x1029485c8 - rayon_core::registry::in_worker::h4d824598a3d5b795
  47:        0x102946a24 - rayon::iter::plumbing::bridge_producer_consumer::helper::h46dc11a49876a36b
  48:        0x10294f540 - <rayon_core::job::StackJob<L,F,R> as rayon_core::job::Job>::execute::h225507a10ea8fa97
  49:        0x10298f5f4 - rayon_core::registry::WorkerThread::wait_until_cold::h42bc99c4c3b80f32
  50:        0x102948648 - rayon_core::registry::in_worker::h4d824598a3d5b795
  51:        0x102946a24 - rayon::iter::plumbing::bridge_producer_consumer::helper::h46dc11a49876a36b
  52:        0x10294f228 - rayon_core::job::StackJob<L,F,R>::run_inline::h6d87b8dede6b685a
  53:        0x102948628 - rayon_core::registry::in_worker::h4d824598a3d5b795
  54:        0x102946a24 - rayon::iter::plumbing::bridge_producer_consumer::helper::h46dc11a49876a36b
  55:        0x10294f540 - <rayon_core::job::StackJob<L,F,R> as rayon_core::job::Job>::execute::h225507a10ea8fa97
  56:        0x10298f5f4 - rayon_core::registry::WorkerThread::wait_until_cold::h42bc99c4c3b80f32
  57:        0x10295fe30 - rayon_core::registry::ThreadBuilder::run::h8ad3583fe401e179
  58:        0x102962b34 - std::sys_common::backtrace::__rust_begin_short_backtrace::ha2f741bf4a09e8b6
  59:        0x1029625f4 - core::ops::function::FnOnce::call_once{{vtable.shim}}::ha1b5dd058be92283
  60:        0x102979294 - std::sys::unix::thread::Thread::new::thread_start::h69a21386d42d2120
  61:        0x19066606c - _pthread_kill
Traceback (most recent call last):
  File "/opt/homebrew/bin/pyrogis", line 8, in <module>
    sys.exit(main())
  File "/opt/homebrew/lib/python3.9/site-packages/pyrogis/__main__.py", line 301, in main
    frames_path = cook_file(input_path, output, parsed_vars)
  File "/opt/homebrew/lib/python3.9/site-packages/pyrogis/__main__.py", line 269, in cook_file
    cooked_dish = cook_dish(path, add_dish_desc, parsed_vars)
  File "/opt/homebrew/lib/python3.9/site-packages/pyrogis/__main__.py", line 122, in cook_dish
    return chef.cook_dish_desc(dish_desc)
  File "/opt/homebrew/lib/python3.9/site-packages/pyrogis/chef/chef.py", line 177, in cook_dish_desc
    return dish.serve()
  File "/opt/homebrew/lib/python3.9/site-packages/pyrogis/ingredients/dish.py", line 98, in serve
    cooked_pixels = recipe.cook(pierogi.pixels)
  File "/opt/homebrew/lib/python3.9/site-packages/pyrogis/ingredients/recipe.py", line 36, in cook
    cooked_pixels = ingredient.cook(under_pixels)
  File "/opt/homebrew/lib/python3.9/site-packages/pyrogis/ingredients/seasonings/threshold.py", line 73, in cook
    cooked_pixels = algorithms.threshold(
pyo3_runtime.PanicException: index out of bounds: the len is 1 but the index is 1
```

I tried this in Python as well (assuming the input below is equivalent to the CLI command above):

in:
```python
import pyrogis
p = pyrogis.Pierogi(file="18deg.png")
t = pyrogis.Threshold(lower_threshold=20,upper_threshold=150)
t.cook(p.pixels)
```

out:
```
thread '<unnamed>' panicked at 'index out of bounds: the len is 1 but the index is 1', src/lib.rs:77:20
note: run with `RUST_BACKTRACE=1` environment variable to display a backtrace
Traceback (most recent call last):
  File "<stdin>", line 1, in <module>
  File "/opt/homebrew/lib/python3.9/site-packages/pyrogis/ingredients/seasonings/threshold.py", line 73, in cook
    cooked_pixels = algorithms.threshold(
pyo3_runtime.PanicException: index out of bounds: the len is 1 but the index is 1
```

Using pyrogis `0.2.0`, MacOS `11.2.3`.

**Comment by ksmoore17** [comment]
Thank you for the bug report. You've got the library usage right too which is cool.

I have a feeling that it's caused by the input pixel array not being in RGB, which is caused by the image not being read in RGB mode (just a bug that could be avoided in code - caused by usage I've never tried). Is this input image greyscale or something funky like that?

There is a 0.3.0 update coming like this week which hopefully helps stability - including a fix for this.
`Pierogi.from_path("18deg.png")` will be the way to do this rather than `Pierogi(file="18deg.png")`

**Comment by useffc** [comment]
Attaching the input image, could be helpful:

<img width="311" alt="18deg" src="https://user-images.githubusercontent.com/1919818/114323835-15e61f80-9af5-11eb-9db1-6391211d47bf.png">

**Comment by ksmoore17** [comment]
Ty... I was able to reproduce it. It seems that image is loading as RGBA by default which doesn't match up with what I understand the error to be saying. Whatever. The threshold on that image looks cool when it works. I will push an update tomorrow evening or the following day that should fix, and I'll notify in this thread.

**Comment by ksmoore17** [comment]
Hi, I believe the 0.3.1 release fixes this. `pip install --upgrade pyrogis`. Let me know if you get a chance

**Comment by useffc** [comment]
addressed in #18
